Mathport/Binary/TranslateName.lean

```diff
@@ -47,14 +48,23 @@ def translateSuffix (s : String) (eKind : ExprKind) (ty : Option Expr) : Binport
     | ExprKind.eDef   => return s.snake2camel
     | ExprKind.eProof => smartName s ty
 
+-- Add `Cat` to the end of a name of a type,
+-- if necessary to disambiguate it from an existing name.
+def addCat (eKind : ExprKind) (n : Name) : BinportM Name := do
+  let rm := Mathlib.Prelude.Rename.getRenameMap (← getEnv)
+  if eKind == .eSort && rm.toLean3.contains n
```

Comment:
This only works if the lean 3 declaration is aligned though, which means it doesn't work at all for declarations that haven't been ported yet.

Reply:
Which is probably fine right? That sounds better than ending up with `Module.EndCat` and `LinearMapCat` in the output of mathport.

Reply:
Once the declaration is aligned, I don't see how the heuristic helps since it doesn't apply anymore.

Reply:
This is intended for the following situation:
```lean
-- lean 3
structure frobnozz (X : Type) : Type := [...]
def Frobnozz : Type := bundled frobnozz
```
I believe what will now happen is that mathport will encounter `frobnozz` and add `frobnozz -> Frobnozz` to its rename map, so that when it encounters `Frobnozz`, it will decide to add `Cat`. This is the case regardless of whether `frobnozz` has been manually aligned to `Frobnozz` as well.

Reply:
@digama0 Scrolling through the output of mathport with this patch suggests that it behaves as one wants in >95% of the cases. The edge cases were posted to zulip in the thread linked to in the top post.